### PR TITLE
[Merged by Bors] - chore: make tests into a library

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: MathlibTest/Lint.lean [hunk 1]
```diff
@@ -7,23 +7,6 @@ import Mathlib.Order.SetNotation
 
 set_option linter.dupNamespace false
 
-namespace termG
```
Comment (kim-em) [inline MathlibTest/Lint.lean:10]: I've removed this (now broken) test on @adomani's [advice](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Making.20Mathlib.20tests.20into.20a.20library/near/479175338).

File: lakefile.lean [hunk 2]
```diff
@@ -65,13 +61,19 @@ package mathlib where
 -/
 
 @[default_target]
-lean_lib Mathlib
+lean_lib Mathlib where
+  leanOptions := mathlibLeanOptions
+  -- Mathlib also enforces these linter options, which are not active by default.
+  moreServerOptions := mathlibOnlyLinters
+  -- Use Batteries' test driver for `lake test`
```
Comment (adomani) [inline lakefile.lean:68]:
```suggestion
```
This is just copy-paste, right?

**Review by mo271 — APPROVED** [review]
Good plan to move tests to a separate library!

**Review by adomani — APPROVED** [review]
I am not a big expert on `lakefile`s, but assuming this passes CI, I am happy with the changes!

File: MathlibTest/CategoryTheory/Bicategory/Normalize.lean [hunk 3]
```diff
@@ -3,6 +3,8 @@ import Mathlib.Tactic.CategoryTheory.Bicategory.Normalize
 open CategoryTheory Mathlib.Tactic BicategoryLike
 open Bicategory
 
+namespace CategoryTheory.Bicategory
+
```
Comment (grunweg) [inline MathlibTest/CategoryTheory/Bicategory/Normalize.lean:7]: Would you like to add the corresponding `end` statement? Not required.

File: MathlibTest/CategoryTheory/Monoidal/Normalize.lean [hunk 4]
```diff
@@ -3,6 +3,8 @@ import Mathlib.Tactic.CategoryTheory.Monoidal.Normalize
 open CategoryTheory Mathlib.Tactic BicategoryLike
 open MonoidalCategory
 
+namespace CategoryTheory.MonoidalCategory
```
Comment (grunweg) [inline MathlibTest/CategoryTheory/Monoidal/Normalize.lean:6]: Would you like to add the corresponding `end` statement? Not required.

**Review by grunweg — APPROVED** [review]
All the changes you made also look good to me. I saw adomani checked on things that might be missing (`test` directories in the code, linters being path-dependent) --- so, with these addressed and CI passing, I'm quite confident the code should work.
Thanks for doing this!

**Review by grunweg — COMMENTED** [review]
CI passes now; the current changes look fine.

I just tried running `lake test` locally, with making a test fail - the output changes now, but in my opinion not for the worse. Having some tracking `[2640/5619] Running MathlibTest.cc` output is a bit weird (given that the number of tests is much smaller) - but having some progress indicator (as opposed to none before) is great.
I do think there is room to iterate on how to present these results, but that need not happen in this PR. So, let me
maintainer merge

**Comment by adomani** [comment]
Does this change mean that we could also provide cache for the tests and replay them locally to see what failed?

**Comment by grunweg** [comment]
I don't know the answer to this question: but if possible, that seems helpful -- and I think this could happen in a follow-up.

**Comment by kim-em** [comment]
That would be great, indeed.

**Comment by grunweg** [comment]
#18550 includes this in the mathlib cache (naturally, untested yet).

**Review by bryangingechen — APPROVED** [review]
Thanks!
bors d+

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**